.github/workflows/build.yml

```diff
@@ -0,0 +1,46 @@
+name: log-exploration-api build
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+
+permissions:
+  contents: read
+  pull-requests: write
+
+jobs:
+
+  build:
+    name: api-build
+    runs-on: ubuntu-20.04
+    
+    steps:
+    - name: Git Checkout
+      uses: actions/checkout@v2
+
+    - name: Install Go
+      uses: actions/setup-go@v2
+      with:
+        go-version: '^1.16.3' # The Go version to download (if necessary) and use.
+    
+    - name: Install dependencies
+      run: |
+        go mod download
+        
+    - name: Run Unit tests
+      run: |
+        go test -race -covermode atomic -coverprofile=covprofile ./pkg/...
```

Comment:
@darecoder can we use our Makefile targets here? Like `make test-unit`

Reply:
Yes, we can. I'll update it to `make test-unit`

Reply:
No, I mean, can we use our `make test-unit` as a command here. Instead of `go test -race ....`